extensions/xcode/src/shared/check-path-or-increment.ts

```diff
@@ -0,0 +1,23 @@
+import { joinPathComponents } from "../shared/join-path-components";
+import * as os from "os";
+import { existsAsync } from "../shared/fs-async";
+
+/**
+ * return a valid file path
+ * @param location: location in which we want to save the playground file.
+ * @param filename: filename of the created or opened playground file.
+ * @param forceCreate: define if we want to force create or not.
+ */
+export async function getValidPath(location: string, filename: string, forceCreate: boolean): Promise<string> {
+  let path = "";
+  let iteration = null;
+  const targetLocation = location.replace(/^~/, os.homedir());
+  do {
+    const dateString = new Date().toISOString().slice(0, 10).replace(/-/g, "");
```

Comment:
Currently the `dateString` is represented in the following format `yyyyMMdd` I guess adding a hyphen between the components would improve the readability like `yyyy-MM-dd`

Example: `MyPlayground-2022-04-22.playground`

Additionally, I guess it would be even better to use the [`toLocaleDateString`](https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleDateString) API of the Date object.